### ISSUE TITLE
Unable to build docker images

### ISSUE THREAD

**Issue by ahsanjav** [gadgetron/gadgetron#1112]
**Describe the bug**
unable to build docker images

**To Reproduce**
Steps to reproduce the behavior, e.g.:
1. clone gadgetron repo
3. from the root of the repository ./build-images.sh --type dev --flavor cuda

**Expected behavior**
I expect it to build the docker container.

**Logs/Screenshots**

```
#8 [gadgetron_baseimage 4/9] RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh     && mkdir -p /opt     && sh miniconda.sh -b -p /opt/conda     && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh     && find /opt/conda/ -follow -type f -name '*.a' -delete     && find /opt/conda/ -follow -type f -name '*.js.map' -delete     && [ -z "4.11.0" ] || /opt/conda/bin/conda install -n base conda=4.11.0     && /opt/conda/bin/conda install -c conda-forge -n base conda-lock     && /opt/conda/bin/conda clean -afy     && groupadd -r conda --gid 900     && usermod -aG conda vscode     && chown -R :conda /opt/conda     && chmod -R g+w /opt/conda     && find /opt -type d | xargs -n 1 chmod g+s
#8 sha256:43d4dd0dae827a65c1b36ebb96e5639a47ba091fb86766bdbc8583ac3678039f
#8 3.763 /bin/sh: 1: /opt/conda/bin/conda: not found
#8 ERROR: executor failed running [/bin/sh -c wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh     && mkdir -p /opt     && sh miniconda.sh -b -p /opt/conda     && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh     && find /opt/conda/ -follow -type f -name '*.a' -delete     && find /opt/conda/ -follow -type f -name '*.js.map' -delete     && [ -z "$CONDA_VERSION" ] || /opt/conda/bin/conda install -n base conda=$CONDA_VERSION     && /opt/conda/bin/conda install -c conda-forge -n base conda-lock     && /opt/conda/bin/conda clean -afy     && groupadd -r conda --gid 900     && usermod -aG conda ${USERNAME}     && chown -R :conda /opt/conda     && chmod -R g+w /opt/conda     && find /opt -type d | xargs -n 1 chmod g+s]: exit code: 127
------
 > [gadgetron_baseimage 4/9] RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh     && mkdir -p /opt     && sh miniconda.sh -b -p /opt/conda     && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh     && find /opt/conda/ -follow -type f -name '*.a' -delete     && find /opt/conda/ -follow -type f -name '*.js.map' -delete     && [ -z "4.11.0" ] || /opt/conda/bin/conda install -n base conda=4.11.0     && /opt/conda/bin/conda install -c conda-forge -n base conda-lock     && /opt/conda/bin/conda clean -afy     && groupadd -r conda --gid 900     && usermod -aG conda vscode     && chown -R :conda /opt/conda     && chmod -R g+w /opt/conda     && find /opt -type d | xargs -n 1 chmod g+s:
------
executor failed running [/bin/sh -c wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh     && mkdir -p /opt     && sh miniconda.sh -b -p /opt/conda     && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh     && find /opt/conda/ -follow -type f -name '*.a' -delete     && find /opt/conda/ -follow -type f -name '*.js.map' -delete     && [ -z "$CONDA_VERSION" ] || /opt/conda/bin/conda install -n base conda=$CONDA_VERSION     && /opt/conda/bin/conda install -c conda-forge -n base conda-lock     && /opt/conda/bin/conda clean -afy     && groupadd -r conda --gid 900     && usermod -aG conda ${USERNAME}     && chown -R :conda /opt/conda     && chmod -R g+w /opt/conda     && find /opt -type d | xargs -n 1 chmod g+s]: exit code: 127
```

**System Information:**
 - OS: [Linux, Ubuntu 20.04]

**Comment by hansenms** [comment]
I am unable to repro this. Just tries. And the specific build step `[gadgetron_baseimage 4/9]` did run. Could it have been a temporary outage at `https://repo.anaconda.com`? Could you try again?

**Comment by ahsanjav** [comment]
Thank you Michael - I have narrowed it down - good news it doesn't seem to be a GT issue - NIH IT updated their network proxy certificates and it's affected conda, wget, pip, etc I have a solution to fix it for our servers but I am not sure how to cleanly reciprocate the behavior inside docker builds without copying the certificates manually which doesn't seem appropriate. This should affect @roopchansinghv and @xueh2 as well. 

If we don't have any clever solutions I will close this issue.

**Comment by hansenms** [comment]
Would it help you to configure Docker to use the proxy: https://docs.docker.com/network/proxy/

**Comment by ahsanjav** [comment]
Thanks Michael. I docker proxy didn't work in this case. The solution seems to be using the NIH specific certificates during the build and pointing conda and pip to the relevant certificates - I may need to keep this out of the main tree. 

I was able to complete the build without memkind - during its install I kept getting a seg-fault . Can someone remind me what that is being used for - Gadgetron compiles just fine without it so I can proceed for our purposes without it for now but would like to figure out why its failing.
